### ISSUE TITLE
Confusion about TFLite suitable models in combination with TF OD API

### ISSUE THREAD

**Issue by Petros626** [tensorflow/tensorflow#56766]
Hello guys,

I would like to know which model exactly from `Model Zoo 1 & 2 ` (https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf1_detection_zoo.md  //// https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2_detection_zoo.md) are suitable for TensorFlow Lite, or can be converted. 

My findings so far are:
1. TFLite only supports `uint8/int8`, so quantized models.
2. it is possible to convert` float32/float16` models to` uint8/int8` models using the converters of TF1&2
3. the Object Detection API supports for TF1&2 **only SSD models**

Questions:
1. when converting models other than SSD model to TFLite model, e.g. faster_rcnn_inception_v2_coco (float) this can only be run on a CPU or GPU, not with TensorFlow Lite (matching point 2 of findings).

I would be really interested in clearing up this mystery to make reliable statements.

**Comment by mohantym** [comment]
Hi @Petros626 !
Could you give it a trying  at this [example notebook](https://colab.sandbox.google.com/gist/mohantym/bebf3ffa30906e30947c2c096926d21f/tensorflow_traffic_sign_detection.ipynb) for the usage of SSD mobilenet with int8 quantization on Edge TPU.  You can modify the code with respect to GPU delegate by referring this [document](https://www.tensorflow.org/lite/performance/gpu). 

Please post a minimal stand alone code for further assistance.
Thank you!

**Comment by Petros626** [comment]
@mohantym  thank you for your answer,

I know the tools how to do that, so your notebook is sure a helpful example how to do it. But the issue is another, maybe you misinterpret my question.

**Comment by Petros626** [comment]
> @Petros626 ! Could you share a minimal stand alone code to explain your findings. Thank you!

@mohantym You misunderstood me, there is no code, no colab or something i try. It's more a theoretical question, where are the framework conditions to this. 
The official answer to use TFLite from TensorFlow is it **supports only SSD models**, right? But I found several other Repos where people, convert for example **Yolo** or **Faster R-CNN** in a tflite format,even it's slower than the SSD family.

**Comment by mohantym** [comment]
Hi @sachinprasadhs ! 
Could you comment on this .
Thank you!

**Comment by Petros626** [comment]
@sachinprasadhs could you clarify this? 

I summarize my findings about this issue.

**Comment by sachinprasadhs** [comment]
Apologies for the delayed response.

- Object detection supports int8 only for SSD TFLite models.
- Object detection supports float for many other models like maskRCNN, FasterRCNN, etc
Hope this clarifies your doubt. Thanks!

**Comment by Petros626** [comment]
So the question would be why `Edge TPU` supports both and the TFLite converter (TF1 - TOCO) I used accepts `uint8`.

I think `uint8` and `int8` are supported by **TensorFlow Lite**. The OD API only allows not to use pretrained models on `float`, because actually the converters outputs errors.

**Comment by sachinprasadhs** [comment]
Some of the Models which are not SSD will also support int8 for example this https://tfhub.dev/tensorflow/efficientdet/lite2/detection/1
The main advantage of int8 models over float is that it offers better inference time, which is crucial for TFLite models.

**Comment by Petros626** [comment]
right EfficientDet is for **TFLite**, but I think some users have tried to use for examplE `Faster R-CNN` in combination with the `graph_rewriter` in the specific `config file` to apply **aware-quantization**, but then fail on converting it from `.pb` file to `.tflite`.

**Comment by pkgoogle** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The TFLite team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the TFLite space.

Thanks.

**Comment by Petros626** [comment]
We can close this issue. I figured out in research the answers, which I was searching for.

**Comment by pkgoogle** [comment]
Hi @Petros626, can you please post your answer when you have a moment? It will likely help the community, thanks!
